metagpt/tools/search_engine_googleapi.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from __future__ import annotations
+
+import asyncio
+import json
+from concurrent import futures
+from urllib.parse import urlparse
+
+import httplib2
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+from metagpt.config import CONFIG
+from metagpt.logs import logger
+
+
+class GoogleAPIWrapper:
+    """Wrapper around GoogleAPI.
+
+    To use this module, you should have the `google-api-python-client` Python package installed
```

Comment:
The pkg doesn't be in `requirement.txt` or do a module exist check.

Reply:
This code is just refactored from metagpt/tools/search_engine.py. Furthermore, it is not used by default, and google-api-python-client is an optional dependency.  There might be a need to rewrite it using aiohttp in the future, as it is not inherently asynchronous.